src/net_processing.cpp

```diff
@@ -2800,10 +2800,9 @@ void ProcessMessage(
             // expensive disk reads, because it will require the peer to
             // actually receive all the data read from disk over the network.
             LogPrint(BCLog::NET, "Peer %d sent us a getblocktxn for a block > %i deep\n", pfrom.GetId(), MAX_BLOCKTXN_DEPTH);
-            CInv inv;
-            inv.type = State(pfrom.GetId())->fWantsCmpctWitness ? MSG_WITNESS_BLOCK : MSG_BLOCK;
-            inv.hash = req.blockhash;
-            pfrom.vRecvGetData.push_back(inv);
+
+            uint32_t inv_type{State(pfrom.GetId())->fWantsCmpctWitness ? MSG_WITNESS_BLOCK : MSG_BLOCK};
+            pfrom.vRecvGetData.emplace_back(CInv(inv_type, req.blockhash));
```

Comment:
This call to `emplace_back()` will create a `CInv` temporary, copy-construct the object in the vector from the temporary and then destroy the temporary. Whereas `emplace_back(inv_type, req.blockhash)` would directly construct the object in the vector with the supplied arguments.

Reply:
Good point, thank you! Done.